### PR TITLE
Implement token search and token modal based on figma

### PR DESCRIPTION
**Pull request by luizakp**
This PR adds:
- Token modal on figma standard
- Token search WIP
- Token search on figma standard

NOTE: token search logic will change to follow the structure presented [here](https://linear.app/bleu-llc/issue/BAL-246/implement-token-search-component). Therefore, the Moralis API and logic will be removed

https://www.loom.com/share/03ec76f0cd174952b5df24aa6b61f25a

### CODE REVIEW

File: packages/next-app/src/hooks/useTransaction.ts [hunk 1]
```diff
@@ -284,11 +284,11 @@ export function useInternalBalancesTransaction({
     const { hash } = data;
     function handleTransactionStatus() {
       if (!hash || !chain) return;
-      const baseTxUrl = getNetworkUrl(chain!.id);
+      const baseTxUrl = getNetworkUrl(chain!.id).url;
       setNotification(
         NOTIFICATION_MAP_INTERNAL_BALANCES[TransactionStatus.SUBMITTING]
       );
-      setTransactionUrl(`${baseTxUrl}${hash}`);
+      setTransactionUrl(`${baseTxUrl}tx/${hash}`);
```
Comment (ribeirojose) [inline packages/next-app/src/hooks/useTransaction.ts:291]: same as above

File: packages/next-app/src/lib/moralis.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { EvmChain } from "@moralisweb3/common-evm-utils";
+import Moralis from "moralis";
+
+Moralis.start({
+  apiKey: process.env.NEXT_PUBLIC_MORALIS_API_KEY as string,
```
Comment (ribeirojose) [inline packages/next-app/src/lib/moralis.ts:5]: `#reminder` add a todo to your task to add this secret to the fly deployment

File: packages/next-app/src/lib/moralis.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}) {
+  const chainMoralis =
+    chain === "Ethereum" ? EvmChain.ETHEREUM : EvmChain.GOERLI;
+
+  const response = await Moralis.EvmApi.token.getWalletTokenBalances({
+    address: userAddress,
+    chain: chainMoralis,
+  });
+
+  return response.toJSON();
```
Comment (ribeirojose) [inline packages/next-app/src/lib/moralis.ts:23]: `#question` does this need to be awaited?

File: packages/next-app/src/components/Input.tsx [hunk 4]
```diff
@@ -19,9 +19,11 @@ export const Input = React.forwardRef<HTMLInputElement, IInput>(
             "w-full selection:color-white box-border inline-flex h-[35px] appearance-none items-center justify-center rounded-[4px] bg-blue4 px-[10px] text-[15px] leading-none text-slate12 shadow-[0_0_0_1px] shadow-blue6 outline-none selection:bg-blue9 hover:shadow-[0_0_0_1px_black] focus:shadow-[0_0_0_2px_black] disabled:bg-blue1"
           )}
         />
-        <div className="h-6 mt-1 text-red-800">
-          {errorMessage && <span>{errorMessage}</span>}
-        </div>
+        {errorMessage && (
+          <div className="h-6 mt-1 text-red-800 text-xs">
```
Comment (ribeirojose) [inline packages/next-app/src/components/Input.tsx:23]: I know this was not added by your changes, but prefer to using [radix colors](https://www.radix-ui.com/colors) here as that should be the convention we've adopted elsewhere in the app

File: packages/next-app/src/components/Dialog.tsx [hunk 5]
```diff
@@ -29,8 +31,9 @@ export function Dialog({
         />
         <DialogPrimitive.Content
           className={cn(
-            "data-[state=open]:animate-contentShow fixed top-[50%] left-[50%] max-h-[85vh] translate-x-[-50%] translate-y-[-50%] rounded-[6px] bg-blue3 p-[25px] shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,_hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] focus:outline-none  overflow-y-auto",
-            customWidth ? customWidth : "w-[90vw] max-w-[450px]"
+            "data-[state=open]:animate-contentShow fixed top-[50%] left-[50%] max-h-[85vh] translate-x-[-50%] translate-y-[-50%] rounded-[6px] bg-blue3 shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,_hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] focus:outline-none",
+            customWidth ? customWidth : "w-[90vw] max-w-[450px]",
```
Comment (ribeirojose) [inline packages/next-app/src/components/Dialog.tsx:35]: could also be 
```suggestion
            { "w-[90vw] max-w-[450px]": customWidth },
```
but I'd argue that if you're taking in `customWidth` I'd probably expect you to pass that width right? 😬

File: packages/next-app/src/components/Table.tsx [hunk 6]
```diff
@@ -3,6 +3,23 @@ import { createContext, useContext } from "react";
 
 const TableContext = createContext({});
 
+const predefinedClasses = {
+  gray: {
+    solid: {
+      dark: "bg-gray-800 border border-gray-700",
```
Comment (ribeirojose) [inline packages/next-app/src/components/Table.tsx]: Same as [this](https://github.com/bleu-studio/balancer-pool-metadata/pull/84/files#r1185004036). Let's use only one source for the color palette definition throughout these apps -- we should probably even add a lint rule to disallow tw colors or remove them altogether if we can -- feel free to log this as a task or take a stab at doing it if you think the time is right

File: packages/next-app/src/components/Table.tsx [hunk 7]
```diff
@@ -3,6 +3,23 @@ import { createContext, useContext } from "react";
...
+    },
+  },
+  blue: {
+    solid: {
+      dark: "bg-blue3",
+    },
+  },
+} as const;
```
Comment (ribeirojose) [inline packages/next-app/src/components/Table.tsx:17]: 🔝

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/PoolMetadataForm.tsx [hunk 8]
```diff
@@ -104,28 +104,30 @@ export default function PoolMetadataForm({
...
         />
       </div>
-      <Input
-        label="Key"
-        placeholder={"Define an attribute key"}
-        disabled={mode === "edit"}
-        {...register("key")}
-        errorMessage={errors.key?.message}
-      />
+      <div className="flex flex-col gap-y-4">
```
Comment (ribeirojose) [inline packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/PoolMetadataForm.tsx:107]: Curious about what motivated this change 👀

File: packages/shared/index.ts [hunk 9]
```diff
@@ -13,10 +13,16 @@ export enum NetworkChainId {
 }
 
 export const networkUrls = {
-  [NetworkChainId.MAINNET]: "https://etherscan.io/tx/",
-  [NetworkChainId.GOERLI]: "https://goerli.etherscan.io/tx/",
-  [NetworkChainId.POLYGON]: "https://polygonscan.com/tx/",
-  [NetworkChainId.ARBITRUM]: "https://arbiscan.io/tx/",
+  [NetworkChainId.MAINNET]: { url: "https://etherscan.io/", name: "Etherscan" },
```
[the remark this answers is not in the document]
Comment (luizakp) [reply packages/shared/index.ts:16]: true, we can think more about it when this sprint ends, WDYT?

File: packages/next-app/src/components/Dialog.tsx (hunk 5 above)
Comment (luizakp) [reply packages/next-app/src/components/Dialog.tsx:35]: yes I'm using like this 

```js
   <Dialog
      content={<TokenModal handleSelectToken={handleSelectToken} />}
      customWidth="w-8/12"
      noPadding
    >

```

File: packages/next-app/src/components/Table.tsx (hunk 6 above)
Comment (luizakp) [reply packages/next-app/src/components/Table.tsx]: Sure! [Here](https://linear.app/bleu-llc/issue/BAL-332/adapt-colors-to-use-only-radixs-palette) 

Also, @mendesfabio could you change the colors on figma to match [Radix's colors](https://www.radix-ui.com/colors)?

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/PoolMetadataForm.tsx (hunk 8 above)
Comment (luizakp) [reply packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/PoolMetadataForm.tsx:107]: The Input component had a hardcoded margin bottom 16px that was specific for pool metadata design. Now the Input has no margin and you have to set for each occasion you want to use. Does that make sense to you?

File: packages/next-app/src/lib/moralis.ts (hunk 3 above)
Comment (luizakp) [reply packages/next-app/src/lib/moralis.ts:23]: No, the response is already awaited, I'm just transforming it in a JSON

File: packages/next-app/src/components/Dialog.tsx (hunk 5 above)
Comment (ribeirojose) [reply packages/next-app/src/components/Dialog.tsx:34]: 🤦 read this too fast, feel free to ignore

File: packages/next-app/src/app/internalmanager/[network]/[operationKind]/[tokenAddress]/page.tsx [hunk 10]
```diff
@@ -56,13 +58,14 @@ export default function Page({
...
     transactionUrl,
+    setSelectedToken,
   } = useInternalBalance();
 
   const { data: internalBalanceTokenData, mutate } = internalBalances
     .gql(chain?.id.toString() || "1")
     .useSingleInternalBalance({
       userAddress: addressLower as `0x${string}`,
-      tokenAddress: params.tokenAddress,
+      tokenAddress: params.tokenAddress, // aqui acho que era pra ser selectedToken.address e refazer p hook toda vez que mudar o token
```
Comment (ribeirojose) [inline packages/next-app/src/app/internalmanager/[network]/[operationKind]/[tokenAddress]/page.tsx:68]: 👁️ 👄 👁️

**Pull request merged**